Sources/StreamChat/Models/Channel+Sorting.swift

```diff
@@ -0,0 +1,166 @@
+                let rhsValue = rhs[keyPath: sort.keyPath]
+
+                if sort.isAscending, evaluate(lhs: lhs, rhs: rhs, isAscending: sort.isAscending) {
+                    return true
+                } else if !sort.isAscending, evaluate(lhs: lhs, rhs: rhs, isAscending: sort.isAscending) {
+                    return true
+                } else {
+                    continue
+                }
+            }
+
+            return false
+        }
+    }
+}
+
+extension ChatChannel {
+    // This method is meant to only be used for sorting functionality.
+    subscript(dynamicMember member: String) -> String? {
+        switch member {
+        case "cid":
+            return cid.rawValue
+        case "name":
+            return name
+        case "type":
+            return type.rawValue
+        case "lastMessageAt":
+            return lastMessageAt?.sortValue
+        case "createdBy":
+            return createdBy?.id
+        case "createdAt":
+            return createdAt.sortValue
+        case "updatedAt":
+            return updatedAt.sortValue
+        case "deletedAt":
+            return deletedAt?.sortValue
+        case "truncatedAt":
+            return truncatedAt?.sortValue
+        case "isHidden":
+            return isHidden.sortValue
+        case "isFrozen":
+            return isFrozen.sortValue
+        case "isMuted":
+            return isMuted.sortValue
+        case "isDeleted":
+            return isDeleted.sortValue
+        case "isDirectMessageChannel":
+            return isDirectMessageChannel.sortValue
+        case "isUnread":
+            return isUnread.sortValue
+        case "memberCount":
+            return memberCount.sortValue
+        case "watcherCount":
+            return watcherCount.sortValue
+        case "membership":
+            return membership?.memberRole.rawValue
+        case "unreadCount":
+            return unreadCount.messages.sortValue
+        case "lastMessageFromCurrentUser":
+            return lastMessageFromCurrentUser?.createdAt.sortValue
+        case "team":
+            return team
+        default:
+            break
+        }
+
+        // Does not work with nested values (eg. "scope.score" -> `extraData: { scope : { score : 1 }}`)
+        if let value = extraData[member] {
+            return value.sortValue
+        }
+
+        return nil
+    }
+}
+
+private extension RawJSON {
+    var sortValue: String? {
+        if let number = numberValue {
+            return number.sortValue
+        } else if let string = stringValue {
+            return string
+        } else if let bool = boolValue {
+            return bool.sortValue
+        } else {
+            return nil
+        }
+    }
+}
+
+private extension Bool {
+    var sortValue: String { self ? "0" : "1" }
+}
+
+private extension Int {
+    var sortValue: String { "\(self)" }
+}
+
+private extension Double {
+    var sortValue: String { "\(self)" }
```

Comment:
Is this okay? Comparing doubles as string? For equality it be okay, but > or < also works?